# Generated by Django 3.1.4 on 2020-12-24 04:42

import django.core.validators
from django.db import migrations, models
import django.db.models.deletion
import re
import usercertmanager.apps.certs.models
import uuid


class Migration(migrations.Migration):

    initial = True

    dependencies = [
    ]

    operations = [
        migrations.CreateModel(
            name='Application',
            fields=[
                ('id', models.UUIDField(default=uuid.UUID('fa7d6481-4009-4747-b2c9-bf394dab8b2f'), editable=False, primary_key=True, serialize=False)),
                ('name', models.CharField(help_text='The name of this application.', max_length=250)),
                ('create_time', models.DateTimeField(auto_now_add=True)),
                ('modify_time', models.DateTimeField(auto_now=True)),
            ],
        ),
        migrations.CreateModel(
            name='Client',
            fields=[
                ('id', models.UUIDField(default=uuid.UUID('5a25921a-200c-43c8-8637-bbc4bd950239'), editable=False, primary_key=True, serialize=False)),
                ('name', models.CharField(help_text='Name of this client', max_length=250)),
                ('create_time', models.DateTimeField(auto_now_add=True)),
                ('modify_time', models.DateTimeField(auto_now=True)),
            ],
        ),
        migrations.CreateModel(
            name='KeyPair',
            fields=[
                ('id', models.UUIDField(default=uuid.UUID('1e3a7209-9ee6-4db0-ae7f-cc1afff4104a'), editable=False, primary_key=True, serialize=False)),
                ('create_time', models.DateTimeField(auto_now_add=True)),
                ('modify_time', models.DateTimeField(auto_now=True)),
                ('public_key', models.TextField(help_text='RSA public key in ASCII format. Do not include the BEGIN and END CERTIFICATE lines', validators=[django.core.validators.RegexValidator(message='This is not a valid key; it contains illegal characters. Ensure that you have removed the BEGIN and END CERTIFICATE lines.', regex=re.compile('^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$', 8)), usercertmanager.apps.certs.models.validate_rsa_cert])),
                ('private_key', models.TextField(help_text='Unencrypted RSA private key in ASCII format. Do not include the BEGIN and END KEY lines', validators=[django.core.validators.RegexValidator(message='This is not a valid key; it contains illegal characters. Ensure that you have removed the BEGIN and END CERTIFICATE lines.', regex=re.compile('^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$', 8)), usercertmanager.apps.certs.models.validate_rsa_key])),
                ('revoked', models.BooleanField(default=False, help_text='Whether this certificate was revoked by this program')),
                ('application', models.ForeignKey(help_text='The application that this key pair is valid for.', on_delete=django.db.models.deletion.CASCADE, to='certs.application')),
                ('client', models.ForeignKey(help_text='The client that this key pair belongs to.', on_delete=django.db.models.deletion.CASCADE, to='certs.client')),
            ],
        ),
    ]
